channeldb/invoice_test.go

```diff
@@ -24,7 +24,7 @@ var (
 	emptyFeatures = lnwire.NewFeatureVector(nil, lnwire.Features)
```

Comment:
Did some testing with Alice running this PR and Bob running master with `--protocol.legacy.onion` set and found that if Bob creates an invoice the Alice will refuse to pay it due to the face that Bob doesnt support the TLV onion option. This is expected but I think that means that we can remove a bunch of logic that switches on `route.Hop.LegacyPayload`:

https://github.com/lightningnetwork/lnd/blob/708bd05df3e573054d56f830fd695628490f52d0/routing/route/route.go#L133-L136

Reply:
Paging @Roasbeef to weigh in on ripping this code out.

Reply:
Have a slight pref for just kinda leaving it in, and also making sure the legacy protocol cfg stuff still works as well to enable, this way less sweeping itest changes are needed. 

So I'm thinking:
  * flip to required 
  * wait
  * observe 
  * ???
  * "enough time passes" 
  * rip out the code at the same time we start to repurpose the feature bits